monai/csrc/filtering/trainable_bilateral/bf_layer_gpu_backward.cu

```diff
@@ -0,0 +1,282 @@
+        for (int i = 0; i < dimensions; i++) {
+          int HalfWindowSizeBack = cHalfWindowSize_arrBack[i]; // Define constant memory as new variable here (!!),
+                                                               // otherwise: cudaErrorMisalignedAddress
+          int neighbourIndex = homeIndex[i] + kernelIndex[i] - HalfWindowSizeBack;
+          int neighbourIndexClamped = min(cSizesBack[i] - 1, max(0, neighbourIndex));
+          if (neighbourIndex != neighbourIndexClamped) {
+            flagNotClamped = false;
+          }
+        }
+
+        scalar_t colorDistance = 0;
+        scalar_t colorDistanceSquared = 0;
+
+#pragma unroll
+        for (int c = 0; c < C; c++) {
+          scalar_t a = inputTensor[batchOffset + neighbourOffset + c * cColorStrideBack];
+          scalar_t b = inputTensor[batchOffset + homeOffset + c * cColorStrideBack]; // Be careful: Here it is (X_k -
+                                                                                     // X_i) and not (X_i - X_q)
+          scalar_t diff = a - b;
+          colorDistance += diff; // Do not take the absolute value here. Be careful with the signs.
+          colorDistanceSquared += diff * diff;
+        }
+
+        scalar_t spatialWeight = gaussianX * gaussianY * gaussianZ;
+        scalar_t colorWeight = exp(cColorExponentConstantBack * colorDistanceSquared);
+        scalar_t totalWeight = spatialWeight * colorWeight;
+
+        // Aggregating values. Only do this if flagNotClamped: Pixels outside the image are disregarded.
+        if (flagNotClamped) {
+          scalar_t filter_kernel_back;
+
+#pragma unroll
+          for (int c = 0; c < C; c++) {
+            // Distinguish cases for k!=i (calculation is done here)
+            // and k==i (partial derivatives are precalculated).
+            // If statement replaces center element of neighborhood/kernel.
+            if (kernelX != cHalfWindowSize_arrBack[0] || kernelY != cHalfWindowSize_arrBack[1] ||
+                kernelZ != cHalfWindowSize_arrBack[2]) {
+              filter_kernel_back = -(1 / outputWeightsTensor[batchOffset + neighbourOffset + c * cColorStrideBack]) *
+                      outputTensor[batchOffset + neighbourOffset + c * cColorStrideBack] * totalWeight * colorDistance /
+                      (cColorSigmaBack * cColorSigmaBack) +
+                  (1 / outputWeightsTensor[batchOffset + neighbourOffset + c * cColorStrideBack]) * totalWeight *
+                      (1 +
+                       inputTensor[batchOffset + homeOffset + c * cColorStrideBack] * colorDistance /
+                           (cColorSigmaBack * cColorSigmaBack)); // inputTensorData[homeOffset] !!
+            } else {
+              filter_kernel_back = dO_dx_ki[batchOffset + homeOffset + c * cColorStrideBack];
+            }
+
+            valueSum += gradientInputTensor[batchOffset + neighbourOffset + c * cColorStrideBack] * filter_kernel_back;
+          }
+
+          weightSum += totalWeight;
```

Comment:
weightSum seems not used in backward path, can be removed?

Reply:
Hi @zehuanw,
Unfortunately, I forgot to sign my commit. However, adding a remediation commit did not work as the DCO still complains. Also, rebasing and searching on Google did not work. Can you help me with signing the commits? Sorry, I am not super familiar with signed commits/DCO. Thank you for your help!

Reply:
Hi @faebstn96 that's fine I'll handle it when merging this PR.